src/utilities/util_dump.c

```diff
@@ -805,15 +842,328 @@ dump_all_records(WT_CURSOR *cursor, bool reverse, bool json)
             if (fputc(',', fp) == EOF)
                 return (util_err(session, EIO, NULL));
         }
-        print_record(cursor, json);
+        WT_RET(print_record(cursor, json));
         once = true;
     }
 
+    if (ret != WT_NOTFOUND)
+        return (util_err(session, ret, reverse ? "WT_CURSOR.prev" : "WT_CURSOR.next"));
+
     if (json && once && fprintf(fp, "\n") < 0)
         return (util_err(session, EIO, NULL));
     return (0);
 }
 
+/*
+ * dump_explore --
+ *     Dump data in an interactive fashion.
+ */
+static int
+dump_explore(WT_CURSOR *cursor, const char *uri, bool reverse, bool pretty, bool hex, bool json)
+{
+    WT_DECL_RET;
+    WT_SESSION *session;
+    uint64_t bookmark_index, window;
+    int i, exact, num_args;
+    char *args[MAX_ARGS], *bookmarks[MAX_BOOKMARKS];
+    char *first_arg, user_input[MAX_ARGS], *current_arg;
+    const char *key, *value;
+    bool once, search_near;
+
+    session = cursor->session;
+    once = search_near = false;
+    i = exact = num_args = 0;
+    bookmark_index = window = 0;
+    for (i = 0; i < MAX_BOOKMARKS; ++i)
+        bookmarks[i] = NULL;
+
+    printf("**************************\n");
+    printf("Explore mode for %s.\n", uri);
+    printf("**************************\n");
+    printf("Enter 'h' for help, 'q' to exit.\n\n");
+
+    while (ret == 0) {
+        i = num_args = 0;
+        fgets(user_input, MAX_ARGS, stdin);
+
+        /* Remove new line character. */
+        user_input[strlen(user_input) - 1] = '\0';
+        if (strlen(user_input) == 0 && !once)
+            continue;
+        once = true;
+
+        /* Time to exit. */
+        if (strlen(user_input) == 1 && user_input[0] == 'q')
+            break;
+
+        /* Parse the input. */
+        current_arg = strtok(user_input, " ");
+        while (current_arg != NULL) {
+            args[i] = malloc(ARG_BUF_SIZE);
+            strcpy(args[i++], current_arg);
+            ++num_args;
+            current_arg = strtok(NULL, " ");
+        }
+
+        first_arg = args[0];
+        switch (first_arg[0]) {
+        /* Cursor info. */
+        case 'a':
+            ret = cursor->get_key(cursor, &key);
+            if (ret != 0 && ret != EINVAL)
+                return (util_cerr(cursor, "get_key", ret));
+            else if (ret == EINVAL) {
+                printf("Error: the cursor needs to be positioned.\n");
+                ret = 0;
+                break;
+            }
+
+            if ((ret = cursor->get_value(cursor, &value)) != 0)
+                return (util_cerr(cursor, "get_value", ret));
+
+            printf("Current position:\n");
+            printf("key:%s\n", key);
+            printf("value:%s\n", value);
+            break;
+        /* Bookmarks. */
+        case 'b':
+            if (strcmp(first_arg, "b") == 0) {
```

Comment:
I don't understand the need for this `strcmp` as `b` is matched in the case stmt.

Reply:
I am checking if this is exactly `b` as we could have, for instance, `bd`.